Sources/Subscription/API/SubscriptionEndpointServiceV2.swift

```diff
@@ -114,7 +114,7 @@ public struct DefaultSubscriptionEndpointServiceV2: SubscriptionEndpointServiceV
             Logger.subscriptionEndpointService.log("Subscription details retrieved successfully: \(subscription.debugDescription, privacy: .public)")
             return try await storeAndAddFeaturesIfNeededTo(subscription: subscription)
         } else {
-            if statusCode == .badRequest {
+            if statusCode == .badRequest || statusCode == .notFound {
```

Comment:
this is the only change in this PR, the rest will disappear as soon the other PRs are merged